tests/st/calicoctl/test_upgrade.py

```diff
@@ -0,0 +1,610 @@
+    'spec': {'interfaceName': 'cali0ef24ba',
+             'ipNetworks': ['fd00:ca:fe:1d:52:bb:e9:80'],
+             'mac': 'ca:fe:1d:52:bb:e9',
+             'profiles': ['profile1']}
+}
+data['do_not_track'] = {
+    'apiVersion': 'v1',
+    'kind': 'policy',
+    'metadata': {'name': 'allow-tcp-555-donottrack'},
+    'spec': {'doNotTrack': True,
+             'ingress': [{'action': 'allow',
+                          'destination': {'ports': [555]},
+                          'protocol': 'tcp',
+                          'source': {'selector': "role == 'cache'"}}],
+             'order': 1230,
+             'selector': "role == 'database'",
+             'types': ['ingress']}
+}
+data['prednat_policy'] = {
+    'apiVersion': 'v1',
+    'kind': 'policy',
+    'metadata': {'name': 'allow-cluster-internal-ingress'},
+    'spec': {'ingress': [{'action': 'allow',
+                          'source': {'nets': ['10.240.0.0/16',
+                                              '192.168.0.0/16']}}],
+             'order': 10,
+             'preDNAT': True,
+             'selector': 'has(host-endpoint)'}
+}
+
+tests = [
+    ("bgppeer_long_node_name", False),
+    ("bgppeer_dotted_asn", False),
+    ("hep_bad_label", True, "a qualified name must consist of alphanumeric characters"),
+    ("hep_tame", False),
+    ("hep_mixed_ip", False),
+    ("hep_label_too_long", True, "name part must be no more than 63 characters"),
+    ("hep_long_fields", False),
+    ("hep_name_too_long", True, "name is too long by 11 bytes"),
+    ("ippool_mixed", False),
+    ("ippool_v4_small", False),
+    ("ippool_v4_large", False),
+    ("node_long_name", False),
+    ("node_tame", False),
+    ("policy_long_name", False),
+    ("policy_big", False),
+    ("policy_tame", False),
+    ("profile_big", False),
+    ("profile_tame", False),
+    ("wep_bad_workload_id", True, "field must not begin with a '-'"),
+    ("wep_lots_ips", False),
+    ("wep_similar_name", True,
+     "workload was not added through the Calico CNI plugin and cannot be converted"),
+    ("wep_similar_name_2", False),
+    ("do_not_track", False),
+    ("prednat_policy", False),
+
+    # profile_long_labels Fails validation after conversion, but error is not clear.
+    # TODO: Add some error text once new validator lands and gives this test a sane error message
+    ("profile_long_labels", True),
+]
+random.shuffle(tests)
+
+
+@parameterized(tests)
+def test_converter(testname, fail_expected, error_text=None):
+    """
+    Convert a v1 object to v3, then apply the result and read it back.
+    """
+    test_converter.__name__ = testname
+    # Let's start every test afresh
+    wipe_etcd(get_ip())
+    testdata = data[testname]
+
+    # Check that the data applies correctly with v2 calicoctl:
+    rc = apply_w_v2_calico(testdata)
+    logger.debug("Trying to apply manifest using V2.6.4 calicoctl.  "
+                 "If this fails, likely to be a script bug")
```

Comment:
which script? test script? because we haven't called the conversion/upgrade script yet.